dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,147 @@
+[[getting-traffic-into-cluster]]
+= Getting Traffic Into The Cluster
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
+== Overview
+There are many ways to access the cluster. This section describes some
+commonly used approaches.
+
+The recommendation is:
+
+- If you have HTTP/HTTPS -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have Encrypted TCP (i.e., TLS with the SNI header) -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have TCP or UDP -- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-externalIP[ExternalIP], or
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-nodeport[NodePort].
+
+
+The admin configures DNS or wildcard DNS records to point to IP addresses
+in the cluster that are exposed outside the cluster.  These addresses are
+commonly the IP addresses of nodes, Virtual IP addresses (VRRP) or Service
+NodePort addreses, Service externalIPs, Service ingressIP, etc..
+
+The admin must make sure the local firewall permits the request to reach
+the IP address.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to make this highly available using replicated services.
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A router is configured
+to accept external reuqests and proxy them based on the configured routes.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+An administrator can create a wild-card DNS entry, and then set up a router,
+after that the users can self-service the ingress without having to bother
+the admins. The router has controls to allow the admin to specify whether
+the users can self-provision hostnames, or if they must fit a pattern the
+admin defines. The other solutions require the admin to do it, or it requires
+that they delegate a lot of privilege.
+
+The admin can create multiple routers and set up router
+xref:../architecture/core_concepts/routes.adoc:#router-sharding[shards]
+
+A set of routes can be created in the various projects. The overall
+set of routes is available to the set of routers. Each router selects
```

Comment:
All routers see all routes unless restricted by labels on the router, this is called router sharding.

The stick the paragraph above as the last sentence of this paragraph.